corporate/lib/stripe.py

```diff
@@ -4699,6 +4699,28 @@ def do_change_remote_server_plan_type(remote_server: RemoteZulipServer, plan_typ
     )
 
 
+@transaction.atomic
+def do_reactivate_remote_server(remote_server: RemoteZulipServer) -> None:
+    """
+    Utility function for reactivating deactivated registrations.
+    """
+
+    if not remote_server.deactivated:
+        billing_logger.warning(
+            "Cannot reactivate remote server with ID %d, server is already active.",
+            remote_server.id,
+        )
+        return
+
+    remote_server.deactivated = False
+    remote_server.save(update_fields=["deactivated"])
+    RemoteZulipServerAuditLog.objects.create(
+        event_type=RealmAuditLog.REMOTE_SERVER_REACTIVATED,
+        server=remote_server,
+        event_time=timezone_now(),
+    )
```

Comment:
Let's add an `acting_user` parameter defaulting to `None` for this function, at least when we hook it up to the support panel (which seems like just a good idea)? Seems fine to merge this for shell use without waiting for that, though.